### PR TITLE
feat: add voice messages

### PR DESCRIPTION
**Pull request by spifory**
## Summary

- Implements discord/discord-api-docs#6082

Tested with: https://paste.nextcord.dev/?id=1725888041228318
image
(the AAAAAA.. was `self._waveform`) 

## This is a **Code Change**

- [x] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

File: nextcord/types/message.py [hunk 1]
```diff
@@ -41,6 +41,8 @@ class Attachment(TypedDict):
     description: NotRequired[str]
     content_type: NotRequired[str]
     spoiler: NotRequired[bool]
+    duration_secs: Optional[float]
+    waveform: Optional[str]
```
Comment (EmreTech) [inline nextcord/types/message.py:45]:
```suggestion
    duration_secs: NotRequired[float]
    waveform: NotRequired[str]
```

File: nextcord/message.py [hunk 2]
```diff
@@ -361,9 +369,22 @@ def to_dict(self) -> AttachmentPayload:
...
 
     @property
+    def waveform(self) -> Optional[bytearray]:
+        """Optional[:class:`str`]: The base64 encoded bytearray representing a sampled waveform
+        (currently for voice messages).
+
+        .. versionadded:: 2.6
+        """
+        if self.waveform is not None:
+            return bytearray(self.waveform)
```
Comment (EmreTech) [inline nextcord/message.py:386]:
```suggestion
            return bytearray(self._waveform)
```

**Review by EmreTech — CHANGES_REQUESTED** [review]

**Review by EmreTech — APPROVED** [review]
Untested.

**Comment by teaishealthy** [comment]
Update versionadded

File: nextcord/message.py [hunk 3]
```diff
@@ -361,9 +371,22 @@ def to_dict(self) -> AttachmentPayload:
...
             result["description"] = self.description
+        if self.duration_secs:
+            result["duration_secs"] = self.duration_secs
+        if self._waveform:
+            result["waveform"] = self._waveform
         return result
 
     @property
+    def waveform(self) -> Optional[array[int]]:
+        """Optional[array[:class:`int`]]: The base64 decoded representing a sampled waveform
```
Comment (teaishealthy) [inline nextcord/message.py:382]:
```suggestion
        """Optional[array[:class:`int`]]: The base64 decoded data representing a sampled waveform
```

**Review by teaishealthy — CHANGES_REQUESTED** [review]

File: nextcord/message.py [hunk 4]
```diff
@@ -362,9 +372,22 @@ def to_dict(self) -> AttachmentPayload:
...
 
     @property
+    def waveform(self) -> Optional[array[int]]:
+        """Optional[array[:class:`int`]]: The base64 decoded data representing a sampled waveform
+        (currently for voice messages).
+
+        .. versionadded:: 3.0
+        """
+        if self._waveform is not None:
+            return array("B", [int(b) for b in base64.b64decode(self._waveform)])
```
Comment (teaishealthy) [inline nextcord/message.py:389]: This is not a cheap conversion, especially with more data. Make this an `utils.cached_property`

**Review by teaishealthy — CHANGES_REQUESTED** [review]

**Comment by teaishealthy** [comment]
Fix the Ruff issues and update your fork

**Comment by Skelmis** [comment]
Any update on this PR?

**Comment by spifory** [comment]
> Any update on this PR?

I forgot about this sorry

File: nextcord/message.py [hunk 5]
```diff
@@ -362,9 +372,23 @@ def to_dict(self) -> AttachmentPayload:
...
-    @property
+    @utils.cached_property
+    def waveform(self) -> Optional[array[int]]:
+        """Optional[array[:class:`int`]]: The base64 decoded data representing a sampled waveform
+        (currently for voice messages).
+
+        .. versionadded:: 3.0
+        """
+        if self._waveform is not None:
+            return array("B", [int(b) for b in base64.b64decode(self._waveform)])
```
Comment (teaishealthy) [inline nextcord/message.py:389]:
```suggestion
            return array("B", base64.b64decode(self._waveform))
```

`base64.b64decode` returns `bytes` which when iterated return `int`

**Review by teaishealthy — CHANGES_REQUESTED** [review]

**Review by Skelmis — APPROVED** [review]
Untested, lines up with expectations

**Comment by Skelmis** [comment]
@teaishealthy care to re-review this?

**Review by vcokltfre — APPROVED** [review]
Tested working. Code looks good.

**Pull request merged**